src/confdb/confdb.c

```diff
@@ -775,6 +776,16 @@ int confdb_init(TALLOC_CTX *mem_ctx,
     int ret = EOK;
     mode_t old_umask;
 
+#ifdef LDB_MODULES_PATH_OVERRIDE
```

Comment:
Since you provide definition of `sss_ldb_init_modules_path()` both in case `LDB_MODULES_PATH_OVERRIDE` is defined and not, there is no need for this additional guard.